ynput/ayon-core
# Bugfix: fix Houdini publish button bugs

## Changelog Description
> Ported from OP https://github.com/ynput/OpenPype/pull/5891

Self publish had few bugs
1. Unnecessary faulty logic in `find_rop_input_dependencies` function which caused karma and HDA publishes to fail.
2. Recursive rop rendering as each extractor was rendering all input nodes! 

Solution:
1. refactor `find_rop_input_dependencies` to `find_rop_inputs_chain` with much simpler logic. 
2. adding `ignore_inputs` flag in `render_rop` function

## Testing notes:
1. use self publish buttons to publish karma and HDA 
2. use different combinations and it should still work
